### PR TITLE
make pulp_pull read from reactor_config

### PR DESCRIPTION
**Pull request by mlangsdorf**
add some helper functions in PulpHandler to get the insecure and secret_path values, and have 
pulp_pull retrieve a PulpHandler from reactor_config and then use those helper values.

### CODE REVIEW

File: tests/plugins/test_pulp_pull.py [hunk 1]
```diff
@@ -155,6 +182,9 @@ def custom_get_v2_no_headers(self, url, headers, **kwargs):
     def custom_get_v2_broken(self, url, headers, **kwargs):
         return self.config_response_config_v2_broken
 
+    """
+    """
```
Comment (bfontecc007) [inline tests/plugins/test_pulp_pull.py]: what's this?

Comment (mlangsdorf) [reply tests/plugins/test_pulp_pull.py]: scaffolding from some debug code.

Removed and fixed.

**Comment by bfontecc007** [comment]
this might need a rebase

**Comment by mlangsdorf** [comment]
Ready for review again.

File: atomic_reactor/plugins/post_pulp_pull.py [hunk 2]
```diff
@@ -60,9 +62,17 @@ def __init__(self, tasker, workflow,
...
+        pulp_config = get_pulp(workflow, False)
+        if pulp_config:
+            pulp_handler = get_pulp_session(workflow, self.log, NO_FALLBACK)
+            self.secret = pulp_handler.get_secret()
+            self.insecure = pulp_handler.get_insecure()
+        else:
+            self.insecure = insecure
+            self.secret = secret
+
+        self.expect_v2schema2 = not get_prefer_schema1_digest(workflow, expect_v2schema2)
```
Comment (lcarva) [inline atomic_reactor/plugins/post_pulp_pull.py]: Won't this negate `expect_v2schema2` as a fallback? Do we want this instead:
```python
self.expect_v2schema2 = not get_prefer_schema1_digest(workflow, not expect_v2schema2)
```

File: atomic_reactor/pulp_util.py [hunk 3]
```diff
@@ -13,11 +13,21 @@
 import warnings
 from collections import namedtuple
 
+try:
```
[the remark this answers is not in the document]
Comment (mlangsdorf) [reply atomic_reactor/pulp_util.py]: Fixed.

File: atomic_reactor/plugins/post_pulp_pull.py (hunk 2 above)
Comment (mlangsdorf) [reply atomic_reactor/plugins/post_pulp_pull.py]: Fixed.

**Comment by lcarva** [comment]
Updated release notes draft:
> - make `pulp_pull` read from reactor config

**Pull request merged**